Tests/Integration/JoinSlowTwinUpdateTests.cs

```diff
@@ -41,30 +41,29 @@ public async Task When_First_Join_Fails_Due_To_Slow_Twin_Update_Retry_Second_Att
                 .ReturnsAsync(twin);
 
             // Device twin will be updated
-            AppSessionKey? afterJoin1AppSKey = null;
-            NetworkSessionKey? afterJoin1NwkSKey = null;
-            string afterJoin1DevAddr = null;
             AppSessionKey? afterJoin2AppSKey = null;
             NetworkSessionKey? afterJoin2NwkSKey = null;
             string afterJoin2DevAddr = null;
 
-            LoRaDeviceClient.Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>()))
-                .ReturnsAsync(true, TimeSpan.FromSeconds(10))
-                .Callback<TwinCollection>((updatedTwin) =>
+            var invocations = 0;
+            LoRaDeviceClient.Setup(x => x.UpdateReportedPropertiesAsync(It.IsNotNull<TwinCollection>(), It.IsAny<CancellationToken>()))
```

Comment:
use SetupSequence instead of trying to keep track of the invocations manually.

Reply:
Added the sequence setup in 9189435